### ISSUE TITLE
Confirm escape & URI encoding path is expected or not in Databricks and Spark

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#15499]
```sql
CREATE TABLE test (c1 int, part varchar) WITH (partitioned_by = array['part']);
INSERT INTO test VALUES (2, 'a%percent');
```

This generates a below partition and "add" entry:
```
_delta_log  part=a%25percent
```

```
{"add":{"path":"part=a%2525percent/20221222_031236_00029_ye7f6-3ca8ac7b-afeb-445d-b2af-1290ef7d603d","partitionValues":{"part":"a%percent"},"size":197,"modificati\
onTime":1671678756868,"dataChange":true,"stats":"{\"numRecords\":1,\"minValues\":{\"c1\":1},\"maxValues\":{\"c1\":1},\"nullCount\":{\"c1\":0}}","tags":{}}}
```

> a%2525percent
1. The path is escaped at io.trino.plugin.deltalake.DeltaLakePageSink#makePartName
2. Then, it's encoded at the io.trino.plugin.deltalake.DeltaLakeMetadata#toUriFormat

We haven't yet observed any issue about this, but it would be better to check the details.

**Comment by findepi** [comment]
Looks like we're percent-encoding twice.

**Comment by findinpath** [comment]
I've investigated the same functionality on Delta Databricks:

```
CREATE TABLE encodingtest (c1 int, part string) 
USING DELTA 
LOCATION 's3://xxxx/encodingtest' 
PARTITIONED BY(part);
```

```
INSERT INTO encodingtest VALUES (2, 'a%percent');
```

This results in having an add file entry in the  transaction log file having this rather peculiar encoding:

```
{"add":{"path":"part=a%2525percent/part-00000-fde5e8ed-6c15-4c2c-833b-2d2c63de91a4.c000.snappy.parquet","partitionValues":{"part":"a%percent"}
```

The same kind of encoding is used when inserting data via Trino:

```
trino> insert into delta.default.encodingtest values (3, 'b%percent');

```
{"add":{"path":"part=b%2525percent/20221229_085145_00002_bnzt3-9e45a5b1-9e32-4816-beb2-6765e12d3897","partitionValues":{"part":"b%percent"},
```

```


Looking into `io.trino.plugin.deltalake.DeltaLakeSplitManager#buildSplitPath`

https://github.com/trinodb/trino/blob/7399e9e418d0b5577809ac4929a7047c4fe07113/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L292-L297

we can see that the `path` is obtained by retrieving the decoded path component of the URI:

https://github.com/trinodb/trino/blob/7399e9e418d0b5577809ac4929a7047c4fe07113/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java#L297

For the `AddFileEntry` with the path `part=a%2525percent/part-00000-fde5e8ed-6c15-4c2c-833b-2d2c63de91a4.c000.snappy.parquet`, 
the  `uri` is `part=a%2525percent/part-00000-fde5e8ed-6c15-4c2c-833b-2d2c63de91a4.c000.snappy.parquet`
and `path` is `part=a%25percent/part-00000-fde5e8ed-6c15-4c2c-833b-2d2c63de91a4.c000.snappy.parquet`

which is in line with what we expect as split path.

See also https://github.com/delta-io/delta/blob/master/PROTOCOL.md#add-file-and-remove-file

I also tested on both Trino and Databricks selecting content from the whole table or from specific partitions and the results matched the expectations.

I consider that this is the intended functionality and not a bug.
